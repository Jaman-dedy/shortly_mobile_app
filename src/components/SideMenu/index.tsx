import React from 'react';
import {StyleSheet} from 'react-native';

import {Box} from '@app/assets/style/theme';
import ShapeImg from '@app/assets/images/shape.svg';
import Button from '@app/components/common/Button';
import Input from '@app/components/common/Input';

const styles = StyleSheet.create({
  shape: {
    alignSelf: 'flex-end',
  },
});

type Props = {
  formData: Record<string, string>;
  errors: Record<string, string>;
  onChangeText: (args: {name: string; value: string}) => void;
  handleShortenLink: () => void;
};

const SideMenu = ({
  formData,
  errors,
  onChangeText,
  handleShortenLink,
}: Props) => {
  console.log('errors :>> ', errors);
  return (
    <Box flex={1} backgroundColor="sideBarBackground">
      <ShapeImg style={styles.shape} />
      <Box position="absolute" alignSelf="center" paddingTop="lg">
        <Input
          onChangeText={onChangeText}
          formData={formData}
          errors={errors}
          placeholder="Shorten a link here..."
        />
        <Button
          text="SHORTEN IT!"
          onPress={handleShortenLink}
          color="primary"
        />
      </Box>
    </Box>
  );
};

export default SideMenu;
