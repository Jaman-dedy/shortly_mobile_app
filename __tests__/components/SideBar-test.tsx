import React from 'react';
import renderer from 'react-test-renderer';
import {cleanup} from '@testing-library/react-native';

import SideMenu from '../../src/components/SideMenu';

import MockThemes from '../../mock/MockTheme';

afterEach(cleanup);
test('renders correctly', async () => {
  const tree = await renderer
    .create(
      <MockThemes>
        <SideMenu />
      </MockThemes>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
