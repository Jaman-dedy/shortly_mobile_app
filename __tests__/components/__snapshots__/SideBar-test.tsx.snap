// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#3B3054",
        "flex": 1,
      },
    ]
  }
>
  <
    style={
      Object {
        "alignSelf": "flex-end",
      }
    }
  />
  <View
    style={
      Array [
        Object {
          "alignSelf": "center",
          "paddingTop": 22,
          "position": "absolute",
        },
      ]
    }
  >
    <TextInput
      onChangeText={[Function]}
      placeholder="Shorten a link here..."
      style={
        Object {
          "backgroundColor": "white",
          "borderColor": "white",
          "borderRadius": 4,
          "borderWidth": 1,
          "color": "black",
          "fontSize": 18,
          "height": 45,
          "paddingHorizontal": 16,
          "paddingVertical": 8,
          "textAlign": "center",
          "width": 280,
        }
      }
      testID="text-input"
    />
    <View
      accessible={true}
      collapsable={false}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#2ACFCF",
          "borderRadiu": 4,
          "marginTop": 8,
          "opacity": 1,
          "paddingHorizontal": 16,
          "paddingVertical": 8,
          "width": 280,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "flexDirection": "row",
              "justifyContent": "center",
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#FFFFFF",
                "fontFamily": "Poppins-Medium",
                "fontSize": 22,
                "fontWeight": "bold",
                "textAlign": "center",
              },
            ]
          }
        >
          SHORTEN IT!
        </Text>
      </View>
    </View>
  </View>
</View>
`;
